source/Util/DurationLogger.ts

```diff
@@ -0,0 +1,29 @@
+import TimerCompletionMessage from './TimerCompletionMessage';
+import { inject, injectable } from 'inversify';
+import TYPES from '../../ioc.types';
+import { Logger } from 'winston';
+import * as moment from 'moment';
+
+@injectable()
+export default class DurationLogger {
```

Comment:
Handy abstraction for differing start and end times - also allows to use a single instance in a nested scenario using a stack. I figured having a nice spot to hide the logic and logging of time diffs would cleanup the code elsewhere and provide more useful information about an operation.